dynamic_programming/all_construct.py

```diff
@@ -0,0 +1,56 @@
+"""
+Program to list all the ways a target string can be
+constructed from the given list of substrings
+"""
+
+
+def all_construct(target: str, word_bank: list[str] = None) -> list[list[str]]:
+    """
+        returns the list containing all the possible
+        combinations a string(target) can be constructed from
+        the given list of substrings(word_bank)
+    >>> all_construct("hello", ["he", "l", "o"])
+    [['he', 'l', 'l', 'o']]
+    >>> all_construct("purple",["purp","p","ur","le","purpl"])
+    [['purp', 'le'], ['p', 'ur', 'p', 'le']]
+    """
+
+    if word_bank is None:
+        word_bank = []
+    # create a table
+    table_size: int = len(target) + 1
+    table: list[list[str]] = []
+    for i in range(table_size):
+        table.append([])
+
+    # seed value
+    table[0] = [[]]  # because empty string has empty combination
```

Comment:
```suggestion
    table: list[list[str]] = [[]] * table_size
    # table[0] will be a seed value because an empty string has an empty combination
```

Reply:
Interestingly , doing this doesn't work ! (despite both initializing the same list 'table' of empty lists) . Not sure why but would recommend to check. And the `table[0] = [[]]` code is mandatory for initializing the seed value required for this dp problem .